includes/constants/class-payment-intent-status.php

```diff
@@ -0,0 +1,31 @@
+<?php
+/**
+ * Class Payment_Intent_Status
+ *
+ * @package WooCommerce\Payments
+ */
+
+namespace WCPay\Constants;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+use MyCLabs\Enum\Enum;
+
+/**
+ * This class gives a list of all Payment Intent status name constants.
+ * ref: https://stripe.com/docs/api/payment_intents/object#payment_intent_object-status
+ *
+ * @psalm-immutable
+ */
+class Payment_Intent_Status extends Enum {
+	const REQUIRES_PAYMENT_METHOD = 'requires_payment_method';
+	const COMPLETED               = 'completed';
```

Comment:
PaymentIntent does not have this status. Might double check again?

Reply:
Hi Dat  @htdat ! Thanks for your review. That was an eagle eye catch! Indeed, my oversight here, the status `Completed` may have crept in from the `Order Status` class that I copied and amended here. 

I checked once again with the [Stripe docs ](https://stripe.com/docs/api/payment_intents/object#payment_intent_object-status)on Payment Intent, and removed the unwanted status. 

I hope this looks fine now?

Reply:
@nagpai - Looks good now! Thanks for the review as well as this whole PR.